lib/p2p/Pool.ts

```diff
@@ -86,19 +104,55 @@ class Pool extends EventEmitter {
   }
 
   /**
-   * Attempt to add an outbound peer by connecting to a given SocketAddress.
-   * Throws an error if a connection already exists for the provided address.
-   * If the connection is successful, send a [[GetHostsPacket]] to discover new XUD hosts to connect to.
+   * Iterate over a collection of nodes and attempt to connect to them.
+   * If the node is banned, already connected, or has no listening addresses, then do nothing.
+   * @param nodes a collection of nodes with a `forEach` iterator to attempt to connect to
+   * @param ignoreKnown whether to ignore nodes we are already aware of, defaults to false
+   * @returns a promise that will resolve when all outbound connections resolve
+   */
+  private connectNodes = (nodes: NodeConnectionIterator, ignoreKnown = false) => {
+    const connectionPromises: Promise<void>[] = [];
+    nodes.forEach((node) => {
+      // check that this node is not ourselves, that it has listening addresses,
+      // and that either we haven't heard of it, or we're not ignoring known nodes and it's not banned
+      if (node.nodePubKey !== this.handshakeData.nodePubKey && node.addresses.length > 0 &&
+        (!this.nodes.has(node.nodePubKey) || (!ignoreKnown && !this.nodes.isBanned(node.nodePubKey)))) {
```

Comment:
Suggestion: all conditions here are negative (including `ignoreKnown`). I think that making them positive will make it easier to follow. So basically checking whether to drop the node, instead of whether to connect to it. `ignoreKnown` can be turned to `allowKnown`,

Reply:
What would that look like? I can't use a `continue` statement here unless I change it from a `foreach`. And fwiw if we are checking whether to skip a node, `ignoreKnown` would make more sense I figure.

I know it's a pretty complex `if` statement, that's why I added the comments above to try to make it more clear.

Reply:
`return` should be the same for that manner.

But you can leave it as is if it doesn't work for you.

Reply:
Oh right `return` would work, I didn't think of that. But I think I'll leave it as is if that's OK. One of the checks does happen to be positive (node.addresses.length > 0).